bash/upload.bash

```diff
@@ -649,16 +698,23 @@ main() {
     _setup_arguments "${@}"
     "${SKIP_INTERNET_CHECK:-_check_internet}"
 
-    [[ -n ${PARALLEL_UPLOAD} ]] && {
-        { command -v mktemp 1>| /dev/null && TMPFILE="$(mktemp -u)"; } || TMPFILE="${PWD}/$(printf "%(%s)T\\n" "-1").LOG"
-    }
+    # create tempfile
+    { command -v mktemp 1>| /dev/null && TMPFILE="$(mktemp -u)"; } || TMPFILE="${PWD}/.$(_t="$(printf "%(%s)T\\n" "-1")" && printf "%s\n" "$((_t * _t))").LOG"
 
     _cleanup() {
         {
-            [[ -n ${PARALLEL_UPLOAD} ]] && rm -f "${TMPFILE:?}"*
+            rm -f "${TMPFILE:?}"*
+
+            # first kill token service pid
+            kill "${ACCESS_TOKEN_SERVICE_PID}"
+
+            # grab and kill all chidren processes of the script
+            # https://askubuntu.com/a/512872
+            # shellcheck disable=SC2086
+            kill $(ps --ppid="$$" -o pid=)
```

Comment:
Codacy found an issue: [Quote this to prevent word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=6363604)